gobblin-metrics-libs/gobblin-metrics/src/main/java/org/apache/gobblin/metrics/OpenTelemetryMetrics.java

```diff
@@ -42,18 +44,26 @@
  * can replace the old metrics system with tighter integrations once it's stable
  */
 
+@Slf4j
 public class OpenTelemetryMetrics extends OpenTelemetryMetricsBase {
 
   private static OpenTelemetryMetrics GLOBAL_INSTANCE;
   private static final Long DEFAULT_OPENTELEMETRY_REPORTING_INTERVAL_MILLIS = 10000L;
+
   private OpenTelemetryMetrics(State state) {
     super(state);
   }
 
   @Override
   protected MetricExporter initializeMetricExporter(State state) {
+    Preconditions.checkArgument(state.contains(ConfigurationKeys.METRICS_REPORTING_OPENTELEMETRY_ENDPOINT),
```

Comment:
wondering: who calls this?  i.e. when the precondition throws, who's getting the exception?

...and are they ready, such that this wouldn't just silently kill some metrics reporting thread w/ no indication other than perhaps a message in our system log?

Reply:
Currently it's called by `KafkaJobStatusMonitorFactory` and it will loudly fail during initialization if this field doesn't exist, good catch